AxonDeepSeg/train_network.py

```diff
@@ -54,588 +58,178 @@ def train_model(path_trainingset, path_model, config, path_model_init=None,
-                    if acc_moving_avg > acc_current_best:
-
-                        save_path = saver.save(session, str(path_model / "best_acc_model.ckpt"))
-                        acc_current_best = acc_moving_avg
-                        print(("Best accuracy model saved in file: %s" % save_path))
-
-                        # Saving the evolution in a pickle file
-                        evolution = {'loss': np.mean(Loss[-10:]),
-                                     'steps': epoch,
-                                     'accuracy': np.mean(Accuracy[-10:])}
-
-                        with open(path_model / 'best_acc_stats.pkl', 'wb') as handle:
-                            pickle.dump(evolution, handle)
-
-                    if loss_moving_avg < loss_current_best:
-
-                        save_path = saver.save(session, str(path_model / "best_loss_model.ckpt"))
-                        loss_current_best = loss_moving_avg
-                        print(("Best loss model saved in file: %s" % save_path))
-
-                        # Saving the evolution in a pickle file
-                        evolution = {'loss': np.mean(Loss[-10:]),
-                                     'steps': epoch,
-                                     'accuracy': np.mean(Accuracy[-10:])}
-
-                        with open(path_model / 'best_loss_stats.pkl', 'wb') as handle:
-                            pickle.dump(evolution, handle)
-
-                # Moreover at a frequency we save the model regardless of the performance
-                if epoch % save_last_epoch_freq == 0:
-
-                    evolution = {'loss': Loss, 'steps': Epoch, 'accuracy': Accuracy}
-                    with open(path_model / 'evolution.pkl', 'wb') as handle:
-                        pickle.dump(evolution, handle)
-                    save_path = saver.save(session, str(path_model / "model.ckpt"))
+    # Data dictionary to feed into image generator
+    data_gen_args = dict(horizontal_flip=True,
+                         # flipping()[1],
+                         vertical_flip=True,
+                         # preprocessing_function = elastic
+                         # flipping()[0],
+                         # rotation_range = random_rotation()[0],
+                         # width_shift_range = shifting(patch_size, n_classes)[1],
+                         # height_shift_range = shifting(patch_size, n_classes) [0],
+                         # fill_mode = "constant",
+                         # cval = 0
+                         )
 
-                    print(("Model saved in file: %s" % save_path))
+    #####################Training###################
+    image_datagen = ImageDataGenerator(**data_gen_args)
+    mask_datagen = ImageDataGenerator(**data_gen_args)
 
-                    evolution = {'loss': np.mean(Loss[-10:]),
-                                 'steps': epoch,
-                                 'accuracy': np.mean(Accuracy[-10:])}
+    seed = 2018
```

Comment:
Here's another hardcoded seed value

Reply:
Seed is the necessary parameter in Keras which if tampered effects performance of the model. May be instead of hardcoded value we can come up with a random value. Is this a viable solution ?

Reply:
Ok, keeping that value is fine then. I wouldn't of expected the random seed to impact performance, but we can keep that fixed value just like it is. I already resolved the conversation, so no change required.